tcms/testruns/static/testruns/js/get.js

```diff
@@ -210,3 +215,47 @@ function fileBugFromExecution(run_id, title_container, container, case_id, execu
 
     dialog.show();
 }
+
+function addLinkToExecution(testExecution) {
+
+    // remove all previous event handlers
+    $('.add-hyperlink-form').off('submit')
+
+    // this handler must be here, because if we bind it when the page is loaded.
+    // we have no way of knowing for what execution ID the form is submitted for.
+    $('.add-hyperlink-form').submit(() => {
+        const url = $('.add-hyperlink-form #id_url').val()
+        const name = $('.add-hyperlink-form #id_name').val()
+        const isDefect = $('.add-hyperlink-form #defectCheckbox').is(':checked')
+        const updateTracker = $('.add-hyperlink-form #autoUpdateCheckbox').is(':checked')
+
+        jsonRPC('TestExecution.add_link', [{
+            execution_id: testExecution.id,
+            url: url,
+            name: name,
+            is_defect: isDefect,
+        }, updateTracker], link => {
+            const ul = $(`.test-execution-${testExecution.id} .test-execution-hyperlinks`)
+            ul.append(renderLink(link))
+
+            // clean the values
+            $('.add-hyperlink-form #id_url').val('')
+            $('.add-hyperlink-form #id_name').val('')
+            $('.add-hyperlink-form #defectCheckbox').bootstrapSwitch('state', false)
+            $('.add-hyperlink-form #autoUpdateCheckbox').bootstrapSwitch('state', false)
+
+            // close the modal
+            $('#add-link-modal button.close').click()
+        })
+
+        return false;
+    })
+
+
+    return true; // so that the modal is opened
+}
+
+function renderLink(link) {
+    const icon = link.is_defect ? `<span class="fa fa-bug"></span>` : ''
+    return `<li>${icon} <a href="${link.url}">${link.name || link.url}</a></li>`
+}
```

Comment:
This should not be like that, we don't want to have any HTML coded inside JavaScript.

Reply:
what is the alternative you suggest?

Reply:
We've discussed this many times already. Use the `<template>` HTML tag, see `<template id="test-execution-row">` in `testruns/get.html` or other usage in `testplans/get.html` (on the refactor_testplan_ui branch).